### DISCUSSION TITLE
Formset Question

### DISCUSSION THREAD

**Discussion by iwalucas** [django-commons/django-unicorn#795] [Q&A] [answered]
I have a django form with an inline formset, and I want to be able to do 2 things, that I cant find a way.
My formset is a list of choices that each have an amount associated, so I want to display the amount and a total at the bottom.

I can't find a way of declaring the formset as an array to manage those relationships.

[This](https://www.django-unicorn.com/docs/django-models/#queryset) could be the answer, but a formset is not a queryset, since I can add/delete/update entries.

Ideas?

**Comment by JohananOppongAmoateng** [comment, marked as answer]
Thanks for the question! Django-unicorn doesn't currently support Django formsets natively (no BaseFormSet, ModelFormSet, or InlineFormSet   handling).        
                                                                                                                       
An alternative is to use a list of dicts as a component property, which gives you the same add/remove/update/total
  functionality:
```
  Component:
  class ChoicesView(UnicornView):
      items = [{"choice": "", "amount": 0}]

      @property
      def total(self):
          return sum(item.get("amount", 0) for item in self.items)

      def add_item(self):
          self.items.append({"choice": "", "amount": 0})

      def remove_item(self, index: int):
          self.items.pop(index)

      def save(self):
          for item in self.items:
              # validate and save here
              ...

  Template:
  {% for item in items %}
    <input unicorn:model="items.{{ forloop.counter0 }}.choice" type="text">
    <input unicorn:model="items.{{ forloop.counter0 }}.amount" type="number">
    <button unicorn:click="remove_item({{ forloop.counter0 }})">Remove</button>
  {% endfor %}

  <button unicorn:click="add_item">Add Row</button>
  <p>Total: {{ total }}</p>
  <button unicorn:click="save">Save</button>
```
